flink-test-utils-parent/flink-connector-test-utils/pom.xml

```diff
@@ -95,4 +95,30 @@
 			<scope>compile</scope>
 		</dependency>
 	</dependencies>
+
+	<build>
+		<plugins>
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-shade-plugin</artifactId>
+				<executions>
+					<execution>
+						<phase>package</phase>
+						<goals>
+							<goal>shade</goal>
+						</goals>
+						<configuration>
+							<shadedArtifactAttached>true</shadedArtifactAttached>
+							<shadedClassifierName>source</shadedClassifierName>
+							<artifactSet>
+								<includes>
+									<include>**/connector/testframe/source/**</include>
```

Comment:
@leonardBang @ruanhang1993 What is this supposed to be?

Reply:
As in, why do we need to create a separate jar when the normal jar would also do the trick?

Reply:
This separate jar is used in the e2e sink tests. It only contains the new [FromElementsSource](https://github.com/apache/flink/pull/18496/files#diff-dbcd767752498ef1c894717f126ffd9008a3c5b20fd3f9a3c6ffefda95cc93d2) which is needed in the sink tests.
The normal jar will contain some other dependencies. I am afraid that it will cause some conflicts. So I create a  separate jar.

As @PatrickRen mentioned, this source could be moved to `flink-streaming-java` in another PR. Then we could get rid of this usage.

Reply:
The normal jar doesn't contain any dependencies though. Unless a module explicitly says to bundle something, no dependencies are bundled.

Reply:
This is reasonable. I will raise a PR to remove this.

Reply:
Thank you!